includes/class-wc-youcanpay-webhook-handler.php

```diff
@@ -38,30 +38,148 @@ public function __construct() {
+            }
+
+            if (!array_key_exists('transaction', $data['payload'])) {
+                return false;
+            }
+
+            if (!array_key_exists('payment_method', $data['payload'])) {
+                return false;
+            }
+
+            $transaction_id = null;
+            $transaction = new WC_YouCanPay_Transaction_Model($data['payload']['transaction']);
+            $payment_method = new WC_YouCanPay_Payment_Method_Model($data['payload']['payment_method']);
+            $payment_method_name = sprintf(__('YouCan Pay (%s)', 'youcan-pay'), $payment_method->get_name());
+
+            if (WC_YouCanPay_Payment_Method_Model::PAYMENT_METHOD_CASH_PLUS != $payment_method->get_name()) {
+                return false;
+            }
+
+            if (!isset($transaction)) {
+                WC_YouCanPay_Logger::info('arrived on process payment: transaction not exists', [
+                    'payment_method' => $payment_method_name,
+                    'code'           => '#0023',
+                    'transaction_id' => $transaction_id,
+                ]);
+
+                WC_YouCanPay_Webhook_State::set_last_webhook_failure_at(time());
+                WC_YouCanPay_Webhook_State::set_last_error_reason(__('The transaction does not exist', 'youcan-pay'));
+
+                return false;
+            }
+
+            $transaction_id = $transaction->get_id();
+            $order = wc_get_order($transaction->get_order_id());
+
+            if (!isset($order)) {
+                WC_YouCanPay_Logger::info('arrived on process payment: order not exists', [
+                    'payment_method' => $payment_method_name,
+                    'code'           => '#0024',
+                    'transaction_id' => $transaction_id,
+                    'order_id'       => $order->get_id(),
+                ]);
+
+                WC_YouCanPay_Webhook_State::set_last_webhook_failure_at(time());
+                WC_YouCanPay_Webhook_State::set_last_error_reason(
+                    __('The order received from the webhook does not exist', 'youcan-pay')
+                );
+
+                return false;
+            }
+
+            if ($transaction->get_status() === WC_YouCanPay_Transaction_Model::PAID_STATUS) {
+                WC_YouCanPay_Logger::info('payment successfully processed', [
+                    'payment_method' => $payment_method_name,
+                    'transaction_id' => $transaction->get_id(),
+                    'order_id'       => $order->get_id(),
+                    'order_total'    => $order->get_total(),
+                ]);
+
+                //TODO: Need to using WC_YouCanPay_Helper::set_payment_method_to_order() instead of $order->set_payment_method()
+
+                $order->set_payment_method($payment_method_name);
+                //WC_YouCanPay_Helper::set_payment_method_to_order( $order, WC_Gateway_YouCanPay::ID );
+                $order->payment_complete($transaction->get_id());
```

Comment:
Apply coding style by using camel case for naming this method.

Reply:
Also, use an action verb when naming methods.  Like completePayment()

Reply:
is a native method of woocommerce